src/System.Management.Automation/engine/SessionStateContainer.cs

```diff
@@ -2382,16 +2382,38 @@ private bool HasGetChildItemDynamicParameters(ProviderInfo providerInfo)
                         // Since the path contained glob characters or we are recursing and the
```

Comment:
@iSazonov, your last commit had 21 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

Enter-PSHostProcess tests.By Process Id.Can enter using NamedPipeConnectionInfo
```powershell
Expected $true, but got $false.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Core\Enter-PSHostProcess.Tests.ps1: line 148
148:                     Should -BeTrue -Because (Get-Content $splat.FilePath -Raw)
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse.Get-ChildItem -Path -Recurse -Hidden
```powershell
Expected the value to have type [System.IO.FileInfo] or any of its subtypes, but got $null with type $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\FileSystemProviderExtended.Tests.ps1: line 175
175:             $result.Where({ $_.Name -eq ".filehidden3.txt" }) | Should -BeOfType System.IO.FileInfo
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse.Get-ChildItem -Path -Recurse -Attributes Hidden
```powershell
Expected the value to have type [System.IO.FileInfo] or any of its subtypes, but got $null with type $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\FileSystemProviderExtended.Tests.ps1: line 184
184:             $result.Where({ $_.Name -eq ".filehidden3.txt" }) | Should -BeOfType System.IO.FileInfo
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse.Get-ChildItem -Path -Recurse -Force
```powershell
Expected the value to have type [System.IO.FileInfo] or any of its subtypes, but got $null with type $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\FileSystemProviderExtended.Tests.ps1: line 193
193:             $result.Where({ $_.Name -eq ".filehidden3.txt" }) | Should -BeOfType System.IO.FileInfo
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse.Get-ChildItem -Path -Recurse -File -Hidden
```powershell
Expected the value to have type [System.IO.FileInfo] or any of its subtypes, but got $null with type $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\FileSystemProviderExtended.Tests.ps1: line 216
216:             $result.Where({ $_.Name -eq ".filehidden3.txt" }) | Should -BeOfType System.IO.FileInfo
```